### PR TITLE
feat: support bulk balance checks on blast

### PR DESCRIPTION
**Pull request by bergeron**
## **Description**

Support bulk balance checks on blast.  This will reduce the number of RPC calls.

I deployed this contract to blast to support the feature:
https://github.com/wbobeirne/eth-balance-checker

This was the tx: https://blastscan.io/tx/0x89a96f29e38ae2c42528ad0b0f07df9465bba6433eda12911c7520bcccc8fbaa

Also verified the source code on blastscan.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23358?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

Switch to blast network, should show native balances correctly across all accounts.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by danjm — DISMISSED** [review]
Looks good to me!

Maybe we should also make a PR against https://github.com/wbobeirne/eth-balance-checker/pull/32

File: app/scripts/constants/contracts.ts [hunk 1]
```diff
@@ -11,4 +11,5 @@ export const SINGLE_CALL_BALANCES_ADDRESSES = {
   [CHAIN_IDS.AVALANCHE]: '0xD023D153a0DFa485130ECFdE2FAA7e612EF94818',
   [CHAIN_IDS.FANTOM]: '0x07f697424ABe762bB808c109860c04eA488ff92B',
   [CHAIN_IDS.ARBITRUM]: '0x151E24A486D7258dd7C33Fb67E4bB01919B7B32c',
+  [CHAIN_IDS.BLAST]: '0xfd5730e96f9dffae40d99b77015bd42816280998',
```
Comment (bergeron) [inline app/scripts/constants/contracts.ts:14]: Future TODO: the bulk balance contract is available on more chains than this list has.  Will add more in a separate PR.

**Comment by bergeron** [comment]
Looking at failing test...

**Review by danjm — APPROVED** [review]

**Comment by bergeron** [comment]
Making sure there's no weird interactions with the fact that blast automatically stakes/rebases ETH.

**Comment by bergeron** [comment]
This is OK to merge from me.  Tested it successfully.

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**